ksqldb-engine/src/main/java/io/confluent/ksql/execution/pull/StreamedRowTranslator.java

```diff
@@ -45,20 +41,12 @@ public StreamedRowTranslator(
   }
 
   @Override
-  public PullQueryRow next() {
-    return rowQueue.pollFirst();
-  }
-
-  @Override
-  public boolean hasNext() {
-    return !rowQueue.isEmpty();
-  }
-
-  public void add(final List<StreamedRow> rows) {
+  public List<PullQueryRow> apply(final List<StreamedRow> rows) {
```

Comment:
Nice.  That rolls the flatmappin' in which probably makes it more sane for others to grok!